### PR TITLE
feat: Add 'Contributing.md' file

### PR DESCRIPTION
**Pull request by Deepakanandrao**
This commit adds the 'Contributing.md' file to provide guidelines for contributing to the project.
It outlines the steps for commit formatting,
pull requests, issue creation, and general community guidelines.

## Changes Made
- Added Contribute.md
- Added .vscode to gitignore

## Issue
[9] (https://github.com/FirecampDev/Firecamp/issues/9)

### CODE REVIEW

**Comment by Deepakanandrao** [comment]
@Nishchit14 IMO, adding some guidelines about the contribution is a good start if the plan is to open source it soon.
Also, you need to allow Pull Requests.

**Comment by Nishchit14** [comment]
That's a great Idea. Thank you for the PR @Deepakanandrao. I am working on integrating the CLA bot then I'll be able to review it for merge.

> Also, you need to allow Pull Requests.

I didn't get it, You have already created the PR. Do you mean creating PR from the same repo instead of forking?

**Comment by Deepakanandrao** [comment]
> That's a great Idea. Thank you for the PR @Deepakanandrao. I am working on integrating the CLA bot then I'll be able to review it for merge.
> 
> > Also, you need to allow Pull Requests.
> 
> I didn't get it, You have already created the PR. Do you mean creating PR from the same repo instead of forking?

No scratch that, looks like bug on my end

File: CONTRIBUTE.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
+# Contributing Guidelines for firecamp.io
+
+Thank you for considering contributing to firecamp.io! To ensure smooth collaboration and maintain a consistent workflow, please follow the guidelines below.
+
+## Commit Guidelines
+
+- We encourage using either the learna or commitzen format for your commit messages.
```
Comment (Nishchit14) [inline CONTRIBUTE.md:7]: should we add the ref links for learna or commitzen?

File: CONTRIBUTE.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+```
+
+Example PR description
+
+```
+## Description
+This pull request adds a new authentication module to improve user security and access control.
+
+## Changes Made
```
Comment (Nishchit14) [inline CONTRIBUTE.md:47]: should we add the PR template also?

File: CONTRIBUTE.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+```
+Bug: Invalid characters in username field
+
+```
+
+Example issue description
+```
+## Issue Description
+When a user enters special characters in the username field during registration, the application throws an error and fails to save the user.
```
Comment (Nishchit14) [inline CONTRIBUTE.md:78]: would be fine to add the issue templates too.

Here is the bug issue template, you can change it inorder to improve.
https://github.com/FirecampDev/Firecamp/blob/main/.github/ISSUE_TEMPLATE/bug_report.md

File: CONTRIBUTE.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+Expected: User should be registered successfully.
+Actual: An error message is displayed, and the user is not saved.
+
+## Screenshots
+![Screenshot 1](/images/screenshot1.png)
+
+
+```
+
+## General Community Guidelines
```
Comment (Nishchit14) [inline CONTRIBUTE.md:94]: would be fine if you can create the `Code of Conduct` template for the same. your thoughts?

**Review by Nishchit14 — CHANGES_REQUESTED** [review]

File: CONTRIBUTE.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
+# Contributing Guidelines for firecamp.io
```
Comment (Nishchit14) [inline CONTRIBUTE.md:1]: The file name should be `CONTRIBUTING.md` as per the [guide](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors)

you can also point the related issue here: https://github.com/FirecampDev/Firecamp/issues/9

File: CONTRIBUTE.md (hunk 4 above)
Comment (Deepakanandrao) [reply CONTRIBUTE.md:94]: 100% We should ideally complete this [checklist] (https://github.com/FirecampDev/Firecamp/community)

File: CONTRIBUTE.md (hunk 1 above)
Comment (Deepakanandrao) [reply CONTRIBUTE.md:7]: `conventional commits` link 👍

**Comment by Nishchit14** [comment]
@Deepakanandrao anyone can create the issue, there is no need for an explicit permission. 
Is this link 404? https://github.com/FirecampDev/Firecamp/issues/new/choose

**Comment by Nishchit14** [comment]
@Deepakanandrao  also are you in a plan to create/update the issue template? if yes then I'll wait or merge this PR for now.

**Comment by Deepakanandrao** [comment]
> @Deepakanandrao also are you in a plan to create/update the issue template? if yes then I'll wait or merge this PR for now.
@Nishchit14 let's update the template in a separate PR against the issue #10

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+- Implemented JWT-based authentication
+- Added user registration and login endpoints
+- Enhanced password encryption and storage
+
+## Screenshots
+![Screenshot 1](/images/screenshot1.png)
+![Screenshot 2](/images/screenshot2.png)
+
+Fixes #123
+
```
Comment (Nishchit14) [inline CONTRIBUTING.md:67]: Is this section required after merging #14? Or we can reference this template. I'm not sure but happy to have your thoughts. 

It's not a big issue, we can keep improving the contributing.md.  @Deepakanandrao

Comment (Deepakanandrao) [reply CONTRIBUTING.md:67]: No, you are right. We can remove the examples and keep the general guidelines from both the sections

**Review by Nishchit14 — APPROVED** [review]

**Pull request merged**